src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -822,7 +825,7 @@ private DataLimits getDataLimits(int userLimit,
         // If we do post ordering we need to get all the results sorted before we can trim them.
         if (aggregationSpec != AggregationSpecification.AGGREGATE_EVERYTHING)
         {
-            if (!needsPostQueryOrdering())
+            if (!needsToSkipUserLimit())
```

Comment:
I think it's the only usage of `needsToSkipUserLimit`, so it can probably be inlined here.